### PR TITLE
Add ECMAScript 6 guidelines doc

### PR DESCRIPTION
**Pull request by JDownloader**
This is a base for what we will use to build the next projects and features using ECMAScript 6 and/or javascript.

Thanks all for your support and kind comments 😍

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -167,8 +167,7 @@ close #47
 ```
 
 ### JavaScript Styleguide
-
-To be sure to respect our guidelines, use this [configuration file](https://github.com/ovh-ux/eslint-config-ovh)
+We are strong supporter of the new features enabled by the ECMAScript 6 language, you can read more about our [javascript guidelines](./JAVASCRIPT_ES6_GUIDELINE.md) to be compliant with our pratices. To be sure to respect our formating and style, use this [configuration file](https://github.com/ovh-ux/eslint-config-ovh)
```
Comment (euhmeuh) [inline .github/CONTRIBUTING.md]: supporterS

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    // ...stuff...
+}
+
+// Good
+function formatUser (userInfos) {
+    // ...stuff...
+    console.log(userInfos.phone);
+}
+```
+Take adventage of destructuring.
```
Comment (euhmeuh) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: advAntage

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    return `<a hreg="${url}"" target="${target}">${url}</a`;
+}
+
+// Allowed
+function (link) {
+    return `<a hreg="${link.url}"" target="${link.target}">${link.url}</a`;
+}
+// Best
+function ({url, target}) {
+    return `<a hreg="${url}"" target="${target}">${url}</a`;
```
Comment (euhmeuh) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: you mean `<a hreF` ?

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+```javascript
+// Bad
+/**
+ * bet() returns a boolean
+ *
+ * @param {Number} i
+ * @return {Boolean} value
+ */
+function bet (i) {
+    let value = flase
```
Comment (AxelPeter) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: typo: false

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (FredericEspiau) [inline .github/CONTRIBUTING.md]: formatting :smiley:

File: .github/JAVASCRIPT_ES6_GUIDELINE.md (hunk 3 above)
Comment (cbourgois) [reply .github/JAVASCRIPT_ES6_GUIDELINE.md]:
```
return `<a href="${url}" target="${target}">${url}</a>`;
```

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+sum === 15;
+
+// Good
+let sum = 0;
+numbers.forEach((num) => {
+  sum += num;
+});
+sum === 15;
+
+// Best (use functional programming to it's advantage)
```
Comment (FredericEspiau) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: :+1:

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+let phrase = "I like " + food + ".";
+
+
+// Bad
+let food = "freshavocado";
+let phrase = ["I like ", food, "."].join();
+
+// Good
+let food = "freshavocado";
+let phrase = `I like ${ food }.`;
```
Comment (FredericEspiau) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: String templates are indented this way right above : `${food}`

File: .github/JAVASCRIPT_ES6_GUIDELINE.md (hunk 4 above)
Comment (JDownloader) [reply .github/JAVASCRIPT_ES6_GUIDELINE.md]: and the semicolon 👍

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 7]
```diff
@@ -72,22 +72,22 @@ function formatUser (userInfos) {
...
 ```
-Take adventage of destructuring.
+Take advantage of destructuring.
 ```
 // Bad
 function (link) {
     let url = link.url;
     let target = link.target;
-    return `<a hreg="${url}"" target="${target}">${url}</a`;
+    return `<a href="${url}" target="${target}">${url}</a`;
```
Comment (cbourgois) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: typo on closing tag </a**>**

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 8]
```diff
@@ -72,22 +72,22 @@ function formatUser (userInfos) {
...
 }
 
 // Allowed
 function (link) {
-    return `<a hreg="${link.url}"" target="${link.target}">${link.url}</a`;
+    return `<a href="${link.url}" target="${link.target}">${link.url}</a`;
```
Comment (cbourgois) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: Same here

File: .github/JAVASCRIPT_ES6_GUIDELINE.md [hunk 9]
```diff
@@ -72,22 +72,22 @@ function formatUser (userInfos) {
...
 }
 // Best
 function ({url, target}) {
-    return `<a hreg="${url}"" target="${target}">${url}</a`;
+    return `<a href="${url}" target="${target}">${url}</a`;
```
Comment (cbourgois) [inline .github/JAVASCRIPT_ES6_GUIDELINE.md]: Same here

File: .github/JAVASCRIPT_ES6_GUIDELINE.md (hunk 7 above)
Comment (JDownloader) [reply .github/JAVASCRIPT_ES6_GUIDELINE.md]: Thanks, now fixed

**Pull request merged**